File: python/exercise32/guess_the_number.py
import random

levels = {1: 10, 2: 100, 3: 1000}
print("Let's play Guess the Number.")
difficulty = input("Pick a difficulty level (1, 2, or 3): ")


try:
    difficulty = int(difficulty)
    if difficulty not in [1, 2, 3]:
        print(f"{difficulty} is not a valid difficulty level!")
        exit(0)
except ValueError:
    print("Invalid difficulty level")
    exit(0)


secret = random.randrange(1, levels[difficulty])
guess_count = 0

while True:
    try:
        print(f"Guess a number between 1 and {levels[difficulty]}!")
        guess = int(input("I have my number. What's your guess? "))
    except ValueError:
        print("Invalid number")
        continue
    finally:
        guess_count += 1

    if guess > secret:
        print("Too high, Guess again")
    elif guess < secret:
        print("Too low, Guess again")
    else:
        print(f"You got it in {guess_count} guesses!")
        break

    print("")

File: go/exercise32/guess_the_number.go
package main

import (
	"bufio"
	"fmt"
	"math/rand"
	"os"
	"strconv"
	"time"
)

var difficulty = map[int]int{
	1: 10,
	2: 100,
	3: 1000,
}

func main() {
	rand.NewSource(time.Now().Unix())
	scanner := bufio.NewScanner(os.Stdin)

	fmt.Println("Let's play Guess the Number.")
	fmt.Print("Pick a difficulty level (1, 2, or 3): ")
	scanner.Scan()

	level, err := strconv.Atoi(scanner.Text())
	if err != nil || level > 3 || level < 1 {
		fmt.Printf("%d is not a valid difficulty level!\n", level)
		os.Exit(0)
	}

	secret := rand.Intn(difficulty[level])
	guessCount := 0

	for {
		fmt.Print("I have my number. What is your guess? ")
		scanner.Scan()
		guess, err := strconv.Atoi(scanner.Text())

		guessCount += 1
		if err != nil {
			fmt.Println("Invalid guess!!")
			continue
		}

		if guess > secret {
			fmt.Println("Too high. Guess again.")
		} else if guess < secret {

			fmt.Println("Too low. Guess again.")
		} else {
			fmt.Printf("You got it in %d guesses.\n", guessCount)
			break
		}
	}
}
